### PR TITLE
Add "Automated Exports" API calls

### PR DESCRIPTION
**Pull request by mohamagdy**
Those API calls are added:

- https://dev.recurly.com/docs/list-export-dates
- https://dev.recurly.com/docs/list-export-files
- https://dev.recurly.com/docs/download-export-file

### CODE REVIEW

**Comment by aaron-suarez** [comment]
Hey there, @mohamagdy! Thanks for your contribution. We'll give this a good review shortly, in the meantime, it looks like Travis is failing on Python 3.5 and below. We have `six` as a dependency to mitigate some of those issues.

**Comment by mohamagdy** [comment]
Thanks @aaron-suarez ... Travis is green now

File: tests/test_resources.py [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-
 from six import StringIO
-from six.moves import urllib, http_client
 from six.moves.urllib.parse import urljoin
 
-from recurly import Account, AddOn, Address, Adjustment, BillingInfo, Coupon, Plan, Redemption, Subscription, SubscriptionAddOn, Transaction, MeasuredUnit, Usage, GiftCard, Delivery, ShippingAddress, AccountAcquisition, Purchase, Invoice, InvoiceCollection, CreditPayment, CustomField
+import recurly
+from recurly import Account, AddOn, Address, Adjustment, BillingInfo, Coupon, Plan, Redemption, Subscription, \
+    SubscriptionAddOn, Transaction, MeasuredUnit, Usage, GiftCard, Delivery, ShippingAddress, AccountAcquisition, \
+    Purchase, Invoice, InvoiceCollection, CreditPayment, CustomField, ExportDate, ExportDateFile
```
Comment (aaron-suarez) [inline tests/test_resources.py]: Thank you for cleaning this up while you were in here

**Comment by bhelx** [comment]
Thank you for picking this up. This is good work!

File: recurly/__init__.py [hunk 2]
```diff
@@ -1504,6 +1504,39 @@ class CreditPayment(Resource):
         'voided_at',
     )
 
+
+class ExportDate(Resource):
+    nodename = 'export_date'
+    collection_path = 'export_dates'
+
+    attributes = ('date', 'export_files')
```
Comment (aaron-suarez) [inline recurly/__init__.py]: Nit: Can you please format this the way the rest of the file formats the `attributes` tuple?
```python
attributes = (
    'date',
    'export_files'
)
```

File: recurly/__init__.py [hunk 3]
```diff
@@ -1504,6 +1504,39 @@ class CreditPayment(Resource):
...
+        """
+        url = urljoin(recurly.base_uri() + self.collection_path, '/%s/export_files' % date)
+        return ExportDateFile.paginated(url)
+
+
+class ExportDateFile(Resource):
+    nodename = 'export_file'
+    collection_path = 'export_files'
+
+    attributes = ('name', 'md5sum', 'expires_at', 'download_url')
```
Comment (aaron-suarez) [inline recurly/__init__.py]: Same nit here

**Comment by aaron-suarez** [comment]
Other than that, I'm perfectly happy with this. We'll merge after that change. Thanks again for contributing!

File: recurly/__init__.py (hunk 2 above)
Comment (mohamagdy) [reply recurly/__init__.py]: Sure, changed in this commit https://github.com/recurly/recurly-client-python/pull/288/commits/1d05a25c3d3fca164886e9c26bd949ce7ce5af44

File: recurly/__init__.py (hunk 3 above)
Comment (mohamagdy) [reply recurly/__init__.py]: Sure, changed in this commit https://github.com/recurly/recurly-client-python/pull/288/commits/1d05a25c3d3fca164886e9c26bd949ce7ce5af44

**Comment by mohamagdy** [comment]
Thanks @aaron-suarez for the review, I did the formatting changes requested.

**Pull request merged**